### PR TITLE
Use string literals in `fixed_point` `release_assert`s

### PR DESCRIPTION
**Pull request by codereport**
`release_assert` is only happy with string literals, not `const car*` and not `std::string`. Solution is to remove `print_rep` and just use a string literal.

### CODE REVIEW

File: cpp/include/cudf/fixed_point/fixed_point.hpp [hunk 1]
```diff
@@ -661,7 +645,7 @@ CUDA_HOST_DEVICE_CALLABLE fixed_point<Rep1, Rad1> operator-(fixed_point<Rep1, Ra
 
   release_assert(
     !subtraction_overflow<Rep1>(lhs.rescaled(scale)._value, rhs.rescaled(scale)._value) &&
-    "fixed_point overflow of underlying representation type " + print_rep<Rep1>());
+    "fixed_point overflow");
```
Comment (jrhemstad) [inline cpp/include/cudf/fixed_point/fixed_point.hpp:648]: wrapping these in `__CUDACC_DEBUG__` kinda defeats the point of _release_ assert. Might as well just use normal `assert`.

**Review by devavret — APPROVED** [review]
What Jake said. And also, thank you.

File: cpp/include/cudf/fixed_point/fixed_point.hpp (hunk 1 above)
Comment (codereport) [reply cpp/include/cudf/fixed_point/fixed_point.hpp]: Done.

**Review by trxcllnt — APPROVED** [review]

**Comment by codereport** [comment]
@gpucibot merge

**Pull request merged**